src/Image.php

```diff
@@ -595,10 +603,60 @@ private function getPathFromResourceName($resourceName)
     public function orientateAndSaveToOriginalPath()
     {
         $img = $this->imageProcess->make($this->getImageResource());
+        switch ($this->orientation) {
+            case 2:
+                $image = $img->flip();
+                break;
+            case 3:
+                $image = $img->rotate(180);
+                break;
+            case 4:
+                $image = $img->rotate(180)->flip();
+                break;
+            case 5:
+                $image = $img->rotate(270)->flip();
+                break;
+            case 6:
+                $image = $img->rotate(270);
+                break;
+            case 7:
+                $image = $img->rotate(90)->flip();
+                break;
+            case 8:
+                $image = $img->rotate(90);
+                break;
+            default:
+                $image = $img;
+        }
+
         $this->outputImage(
-            $this->getCreatedImageResource($img->orientate()),
+            $this->getCreatedImageResource($image),
             $this->getImagePath()->getOriginalPath()
         );
         return $this;
     }
+
+    /**
+     * @return void
+     */
+    private function extractExifDataAndSetOrientation()
+    {
+        if ($this->getMimeTypeFromBase64() === Consts::IMAGE_JPEG) {
+            $this->orientation = !empty(exif_read_data($this->base64Encoded)['Orientation'])
```

Comment:
Extract `exif_read_data($this->base64Encoded)['Orientation']` to a variable (or `exif_read_data($this->base64Encoded))` because you have a double call to `exif_read_data`